pkg/kubernetes/podhelper/containers.go

```diff
@@ -54,6 +51,50 @@ func (check EnvoySidecarImageCheck) FixIt() error {
 	panic("implement me")
 }
 
+// Verify interface compliance
+var _ common.Runnable = (*OsmInitContainerImageCheck)(nil)
+
+// OsmInitContainerImageCheck implements common.Runnable
+type OsmInitContainerImageCheck struct {
+	cfg configurator.Configurator
+	pod *v1.Pod
+}
+
+// HasExpectedOsmInitImage checks whether a pod has a sidecar with the osm init container image specified in the meshconfig
+func HasExpectedOsmInitImage(osmConfigurator configurator.Configurator, pod *v1.Pod) OsmInitContainerImageCheck {
+	return OsmInitContainerImageCheck{
+		cfg: osmConfigurator,
+		pod: pod,
+	}
+}
+
+// Info implements common.Runnable
+func (check OsmInitContainerImageCheck) Info() string {
+	return fmt.Sprintf("Checking whether pod %s has a container with osm init image matching meshconfig init container image", check.pod.Name)
+}
+
+// Run implements common.Runnable
+func (check OsmInitContainerImageCheck) Run() error {
+	for _, container := range check.pod.Spec.InitContainers {
+		fmt.Printf("Container image for pod %s is %s", check.pod.Name, container.Image)
+		fmt.Printf("Expected image for pod %s is %s", check.pod.Name, check.cfg.GetInitContainerImage())
+		if container.Image == check.cfg.GetInitContainerImage() {
+			return nil
+		}
+	}
+	return ErrExpectedOsmInitImageMissing
+}
```

Comment:
This will cause the `fmt.Printf` to be printed a few times with repeated messages. Let's have it this way instead:

```
func (check OsmInitContainerImageCheck) Run() error {
	fmt.Printf("Expected image for pod %s is %s", check.pod.Name, check.cfg.GetInitContainerImage())
	for _, container := range check.pod.Spec.InitContainers {
		fmt.Printf("Container image for pod %s is %s", check.pod.Name, container.Image) // You decide if this line is worth printing...
		if container.Image == check.cfg.GetInitContainerImage() {
			return nil
		}
	}
	return ErrExpectedOsmInitImageMissing
}
```

Reply:
oops those were for debugging, meant to remove them entirely actually